klaviyo/public.py

```diff
@@ -63,8 +71,27 @@ def track(
         if ip_address:
             params['ip'] = ip_address
 
-        query_string = self._build_query_string(params, is_test)
-        return self._public_request(self.TRACK, query_string)
+        if method.lower() not in [KlaviyoAPI.HTTP_POST, KlaviyoAPI.HTTP_GET]:
+
+            raise KlaviyoException('method argument must be either "post" or "get"')
+
+        if method == KlaviyoAPI.HTTP_POST:
```

Comment:
I'd propose forcing method to lowercase and saving as a variable so you can use it here and in the above validation. as it stands you won't get into this clause if someone passed `"POST"`  as the method